### PR TITLE
Changes in the output handling and some configs

### PR DESCRIPTION
**Pull request by ceiphas**
- Added config values for defaults: extra seed chance, essence drop chance, shard crafting recipe
- Changed the handling of the output from three hard coded recipes to an open format, supporting unlimited recipes per shard

the changes were tested with old and new seed files, and should be fully backward compatible

### CODE REVIEW

**Comment by TehNut** [comment]
After a once-over on my phone, this looks good. Will look over a bit more when I get home

File: src/main/java/tehnut/resourceful/crops/api/base/Output.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import net.minecraft.item.ItemStack;
+import tehnut.resourceful.crops.ConfigHandler;
+import tehnut.resourceful.crops.api.ResourcefulAPI;
+
+@EqualsAndHashCode
+public class Output {
+    private ItemStack            outputStack;
+    private String               recipe;
+    private static final Pattern recipePattern = Pattern.compile("^[S #]{5,11}$");
+    private static final Pattern linePattern   = Pattern.compile("^[S ]{2,3}$");
```
Comment (TehNut) [inline src/main/java/tehnut/resourceful/crops/api/base/Output.java]: Not a fan of the whitespacing for these fields

File: src/main/java/tehnut/resourceful/crops/api/base/OutputBuilder.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Factory for creating SeedReqs. Documentation for each field can be found in
+ * {@link Requirement}
+ */
+@Setter
+@Accessors(chain = true)
+@NoArgsConstructor
+public class OutputBuilder {
+    private ItemStack outputStack;
+    private String    recipe = "default";
```
Comment (TehNut) [inline src/main/java/tehnut/resourceful/crops/api/base/OutputBuilder.java]: Whitespacing

File: src/main/java/tehnut/resourceful/crops/registry/RecipeRegistry.java [hunk 3]
```diff
@@ -29,30 +30,20 @@ public static void registerItemRecipes() {
...
-                    GameRegistry.addRecipe(new ShapedOreRecipe(
-                            seed.getThirdOutput(),
-                            "SS", "SS",
-                            'S', new ItemStack(ItemHelper.getItem(ItemShard.class), 1, ResourcefulAPI.SEEDS.getId(seed)))
-                    );
-                }
+	                if (seed.getOutput() != null) {
+	                	for (Output output : seed.getOutput()) {
+			            	try {
+	                		GameRegistry.addRecipe(new ShapedOreRecipe(
```
Comment (TehNut) [inline src/main/java/tehnut/resourceful/crops/registry/RecipeRegistry.java]: Spacing

File: src/main/java/tehnut/resourceful/crops/api/base/Output.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            ResourcefulAPI.logger.info("No recipe found, using 'default': " + this.toString());
+            return translateRecipe("default");
+        }
+        switch (recipe.length) {
+            case 1:
+                if (recipe[0] == null) {
+                    ResourcefulAPI.logger.info("No recipe found, using 'default': " + this.toString());
+                    return translateRecipe("default");
+                }
+                if ("default".equals(recipe[0])) {
```
Comment (TehNut) [inline src/main/java/tehnut/resourceful/crops/api/base/Output.java]: How about making these an enum and using `Enum#valueOf(recipe [0]).getRecipe()`

**Review by TehNut — CHANGES_REQUESTED** [review]

File: src/main/java/tehnut/resourceful/crops/api/base/Output.java (hunk 4 above)
Comment (ceiphas) [reply src/main/java/tehnut/resourceful/crops/api/base/Output.java]: Problem is, that 2x2 is not a valid Enum value, as is 3x3...

File: src/main/java/tehnut/resourceful/crops/api/base/Output.java (hunk 1 above)
Comment (ceiphas) [reply src/main/java/tehnut/resourceful/crops/api/base/Output.java]: sorry, i'll tell my eclipse to remove that

File: src/main/java/tehnut/resourceful/crops/api/base/OutputBuilder.java (hunk 2 above)
Comment (ceiphas) [reply src/main/java/tehnut/resourceful/crops/api/base/OutputBuilder.java]: sorry, i'll tell my eclipse to remove that

**Review by TehNut — APPROVED** [review]

**Comment by TehNut** [comment]
I pushed some commits I forgot I had locally. If you resolve the conflicts, I will merge. Sorry!

**Pull request merged**